update_observed_aimpoints.py

```diff
@@ -340,35 +338,48 @@ def plot_observed_aimpoints(obs_aimpoints):
-    plot_cxctime(times[~ok], obs_aimpoints['dx'][~ok], '*b', label='CHIPX (offset > 100")')
-    plot_cxctime(times[~ok], obs_aimpoints['dy'][~ok], '*r', label='CHIPY (offset > 100")')
 
-    for axis in ('dx', 'dy'):
-        if np.any(lolims[axis]):
-            plt.errorbar(DateTime(times[lolims[axis]]).plotdate,
-                         obs_aimpoints[axis][lolims[axis]], marker='.', yerr=1.5, lolims=True)
-        if np.any(uplims[axis]):
-            plt.errorbar(DateTime(times[uplims[axis]]).plotdate,
-                         obs_aimpoints[axis][uplims[axis]], marker='.', yerr=1.5, uplims=True)
-
-    plt.grid()
-    ymax = max(12, np.max(np.abs(obs_aimpoints['dx'])), np.max(np.abs(obs_aimpoints['dy'])))
-    plt.ylim(-ymax, ymax)
-    plt.ylabel('Offset (arcsec)')
-    plt.title('Observed aimpoint offsets')
-
-    plt.legend(loc='upper left', fontsize='small', title='', framealpha=0.5)
-
-    outroot = os.path.join(opt.data_root, 'observed_aimpoints')
-    logger.info('Writing plot files {}.png,html'.format(outroot))
-    mpld3.plugins.connect(fig, mpld3.plugins.MousePosition(fmt='.1f'))
-    mpld3.save_html(fig, outroot + '.html')
-    fig.patch.set_visible(False)
-    plt.savefig(outroot + '.png', frameon=False)
+    for idx, axis, label in zip([1, 2], ['dx', 'dy'], ['CHIPX', 'CHIPY']):
+        plt.close(idx)
+        fig = plt.figure(idx, figsize=(8, 4))
+
+        for det, c in zip(['HRC-I', 'HRC-S', 'ACIS-I', 'ACIS-S'],
+                          ['cyan', 'magenta', 'red', 'blue']):
+            ok = ((np.abs(obs_aimpoints['target_offset_y']) < 100) &
+                  (np.abs(obs_aimpoints['target_offset_z']) < 100) &
+                  (obs_aimpoints['detector'] == det))
+            nok = ((np.abs(obs_aimpoints['target_offset_y']) >= 100) &
+                  (np.abs(obs_aimpoints['target_offset_z']) >= 100) &
+                  (obs_aimpoints['detector'] == det))
+
+            if np.count_nonzero(ok):
+                plot_cxctime(times[ok], obs_aimpoints[axis][ok], marker='o', color=c, linestyle='', alpha=.5,
+                             label='{}'.format(det))
+            if np.count_nonzero(nok):
+                plot_cxctime(times[nok], obs_aimpoints[axis][nok], marker='*', color=c, linestyle='',
+                             label='{} (offset > 100")'.format(det))
+            if np.any(lolims[axis]):
+                plt.errorbar(DateTime(times[lolims[axis]]).plotdate,
+                             obs_aimpoints[axis][lolims[axis]], marker='.', linestyle='',
+                             color=c, yerr=1.5, lolims=True)
+            if np.any(uplims[axis]):
+                plt.errorbar(DateTime(times[uplims[axis]]).plotdate,
+                             obs_aimpoints[axis][uplims[axis]], marker='.', linestyle='',
+                             color=c, yerr=1.5, uplims=True)
+        plt.grid()
+        ymax = max(12, np.max(np.abs(obs_aimpoints[axis])))
+        plt.ylim(-ymax - 5, ymax + 5)
```

Comment:
I'm fine with opening up the limits.  But at this point allowing for smaller limits is probably not that useful, it is better to have a consistent plot with always the same scale.  So just hardwire to +/- 17.  You might as well increase the out-of-bounds limit from 10 to 15 while you are at it.